ouroboros-consensus-diffusion/app/conformance-test-runner/Main.hs

```diff
@@ -23,9 +28,27 @@ import Ouroboros.Network.OrphanInstances ()
 import Ouroboros.Network.PeerSelection (PeerAdvertise (..), PortNumber)
 import Ouroboros.Network.PeerSelection.LedgerPeers
 import Ouroboros.Network.PeerSelection.State.LocalRootPeers (HotValency (..), WarmValency (..))
+import System.Exit (ExitCode (..))
 import Test.Consensus.PointSchedule
 import Test.Consensus.PointSchedule.Peers (PeerId (..), Peers (Peers), getPeerIds)
 
+data ExitStatus = InternalError | BadUsage | Flags (Set StatusFlag)
+
+data StatusFlag = TestFailed | ContinueShrinking deriving (Eq, Ord)
+
+exitStatusToCode :: ExitStatus -> ExitCode
+exitStatusToCode = \case
+  InternalError -> ExitFailure (bit 0)
+  BadUsage -> ExitFailure (bit 1)
+  Flags flags ->
+    if null flags
```

Comment:
as a nit, I greatly prefer `case` over `if`. No sense in keeping around special syntax for monomorphic concepts.